doc/west/index.rst

```diff
@@ -1,14 +1,122 @@
+* Out-of-tree development based on subsets of the supported boards and SoCs
+
+Command-line interface
+======================
+
+Zephyr uses CMake as its build system management tool, as described in
+:ref:`application`.
+While CMake is a very powerful sytem and language, its two-stage nature
+as a build system generator makes it non-trivial to present the user with
+a consistent set of command-line switches and options that enable advanced
+usage of the build tool ultimately responsible for building, flashing and
+debugging the application.
+In order to unify and streamline the developer's experience when working with
+Zephyr, west was introduced. It is very important to note that west does not
+replace the CMake-based build system, but directly uses it and augments it.
+
+.. _west-struct:
+
+Structure
+*********
+
+West is downloaded and installed on a system in two stages:
+
+* Bootstrapper: Installed using ``pip``, implements ``west init``
+* Installation: Installed using ``west init``, implements all other commands
+
+Additional information about the two parts can be found below.
+
+Bootstrapper
+============
+
+The bootstrapper module is distributed using `PyPi`_ (i.e. :file:`pip`) and it's
+placed in the user's ``PATH`` becoming the only entry point to west.
+It implements a single command: ``west init``. This command needs to be run
+first in order to use the rest of functionality included in ``west``, since
+``west init`` will do the following:
+
+* Clone west itself in a :file:`west/west` folder
+* Clone the manifest in a :file:`west/manifest` folder (additional information
+  on the manifest can be found in the :ref:`west-multi-repo` section)
+
+Once ``west init`` has been run, the bootstrapper will delegate the handling of
+any commands different from ``init`` to the cloned west installation.
+
+This means that there is a single bootstrapper instance installed at any time
+(unless you use virtual environments), which can then be used to initialize as
+many installations as needed.
+
+.. _west-struct-installation:
+
+Installation
+============
+
+A west installation is the result of running ``west init`` on a folder. As
+described on the previous section, an installation always includes a special
+:file:`west/` folder that contains a clone of both west itself and the manifest.
+Additionally, and once ``west clone`` has been run, the installation will also
+contain a clone of any repositories listed in the manifest. The directory tree
+layout of an installation using the default Zephyr manifest therefore looks
+like this:
+
+.. code-block:: console
+
+   ├── west
+   │   ├── manifest
+   │   └── west
+   └── zephyr
+
+Additionally, ``west init`` creates a configuration file :file:`west/config`
+that stores configuration metadata about the west installation.
+
+To learn more about west's handling of multiple repositories refer to
+:ref:`west-multi-repo`.
+
+All west commands except ``west init`` are implemented by the :file:`west/west`
```

Comment:
Much clearer, thanks.

I could just be hearing cacofonía where there isn't any, so feel free to ignore, but what about:

```
The west bootstrapper delegates handling for all commands except ``init`` to the installation in :file:`west/west`.
```

Reply:
Nice, I like it thanks!